### PR TITLE
prepass globals binding -> 9

### PR DESCRIPTION
**Pull request by robtfm**
# Objective

in pbr prepass, the globals binding is bound to slot 1, whereas in main pass it is bound to slot 9. the import (mesh_view_bindings.wgsl) also assumes slot 9.

## Solution

move it to slot 9

## Migration Guide

if you manually import the `global` binding in a prepass fragment shader, change your binding slot to 9, or alternatively `#import bevy_pbr::mesh_view_bindings::globals` directly.

### CODE REVIEW

**Comment by robtfm** [comment]
this is a bit scrappy. it requires importing the full set of view bindings just to use the globals declaration (none of the others will be available if users try to access them), and relies on tree-shaking (from naga_oil, nothing vendor specific) to exclude the other unused bindings to avoid conflicts.

i intend to rework these bindings soon anyway, so for now i think it's ok.

File: crates/bevy_pbr/src/prepass/mod.rs [hunk 1]
```diff
@@ -690,17 +690,20 @@ pub fn prepare_prepass_view_bind_group<M: Material>(
...
                 "prepass_view_motion_vectors_bind_group",
                 &prepass_pipeline.view_layout_motion_vectors,
-                &BindGroupEntries::sequential((
-                    view_binding,
-                    globals_binding,
-                    previous_view_proj_binding,
+                &BindGroupEntries::with_indices((
+                    (0, view_binding),
+                    (9, globals_binding),
+                    (2, previous_view_proj_binding),
```
Comment (superdump) [inline crates/bevy_pbr/src/prepass/mod.rs:706]: I feel like I'd prefer these in ascending order. :)

**Review by superdump — CHANGES_REQUESTED** [review]

**Comment by IceSentry** [comment]
Should this be in 0.12.1? A user recently mentioned having issue migrating a prepass shader that used globals and this was probably the issue.

**Review by IceSentry — APPROVED** [review]

**Comment by robtfm** [comment]
I didn’t realise it had changed since previous versions, if so then yes. Otherwise it might be considered breaking so maybe not suitable for a patch? 

In particular Chris biscardi used a manual binding at slot 1 in the rust graphics weekly video on extended materials, it would be a bit of a shame to break that so quickly.

**Comment by IceSentry** [comment]
Okay, I just looked at the issue again. So the issue is that the binding on the shader side was removed. As in, you can't do `#import bevy_pbr::prepass_bindings::globals` anymore, but adding the binding declaration manually should still work. Here's the issue in question https://github.com/bevyengine/bevy/issues/10652

I guess the fix would be to reintroduce the shader side bindings. I'm not sure why they were removed.

**Comment by IceSentry** [comment]
New fix: https://github.com/bevyengine/bevy/pull/10667

**Pull request closed without merging**